### ISSUE TITLE
Can't define state in subclass

### ISSUE THREAD

**Issue by mikelambert** [facebook/flow#1919]
The following code gives an error:

```
/**
 * @flow
 */
import React from 'react';
class ParentClass extends React.Component {
}
export class ChildClass extends ParentClass {
  state: {
    addedBy: ?string,
  };
}
```

Gives me the following error:

```
test.js:7
  7: class ParentClass extends React.Component {
                               ^^^^^^^^^^^^^^^ undefined. Did you forget to declare type parameter `State` of property `Component`?. This type is incompatible with
 12:   state: {
              ^ object type

test.js:12
 12:   state: {
              ^ object type. This type is incompatible with
  7: class ParentClass extends React.Component {
                               ^^^^^^^^^^^^^^^ undefined. Did you forget to declare type parameter `State` of property `Component`?
```

Is defining state in the subclasses where it is necessary supported? It works fine with React Native, but just fails with Flow.

**Comment by asolove** [comment]
This is now fixed in v0.53. The only change you need to make is asking flow to infer the type variables on `Component`:

```
class ParentClass extends React.Component<*,*>
```

And then this example typechecks.

(@calebmer this can be closed)

**Comment by mikelambert** [comment]
I've since changed my design to use composition instead of inheritance to avoid the type errors, so don't have an easy test case to verify with. But I'll trust you, and mark this closed. Thanks!
